### DISCUSSION TITLE
Possible to make overlays not take focus when clicked ?

### DISCUSSION THREAD

**Discussion by dlm21** [xpdota/event-trigger#349] [Q&A]
On Linux, with Wayland, I noticed that if I click on any of the overlays, it steals input focus from the game, so for example if I need to move and right click on the ground which has a transparent overlay in that spot, none of my keyboard inputs make it to the game, which is super dangerous in a lot of content. I can click back on the game where there's not an overlay, but I've found myself doing it a lot on the callouts overlay (where I need it right in the middle) quite a lot.
I know some overlays in other projects have a mode where it passes through all clicks/inputs, and you have to toggle that mode off if you want to interact with/move the overlay. Is that possible with this ?

**Comment by xpdota** [comment]
The overlays *should* be click-through already. I wonder if that doesn't work right on Wayland? The native call responsible for that is is `XShapeCombineMask`, but I can't find much information about whether it works on Wayland or if there's a WL equivalent.

**Reply by dlm21** [reply]
Just to clarify, the mouse CAN click through the overlays, I can right click on the ground in the game behind/through and overlay, spin the camera, etc, but that appears to be raising focus so that keyboard input is no longer going to the game. For example, if I'm holding "w" to go forward, then left or right click on a spot with an overlay, my character stops moving, even though I'm still holding "w". I'll test in Xorg after work tonight just to see if it's a X vs Wayland issue and report back.

**Reply by dlm21** [reply]
Ok, so on KDE Plasma w/ X11, the overlays don't interfere and I can play uninterrupted by them. Tested again on KDE Plasma w/ Wayland and it was having the issue previously described, so all signs point to being a Wayland issue.  

Note I can replicate the issue even out of the game, just open a terminal underneath an overlay area, hold down a key on the keyboard then left or right click where that overlay is, and you can see the input to the term stop. 

I also searched around for info on it, but it appears XShapeCombineMask is only a thing in X11, so perhaps the bug is in XWayland  where I assume java is being "routed" through. Ah well, I'll move all my overlays to the left side of the screen for now as a workaround, if I run across anything that might help or notice it gets fixed from upstream changes, I'll pass it along. Thanks again for taking a look.
